otherlibs/str/str.mli

```diff
@@ -50,21 +50,20 @@ val regexp : string -> regexp
    - [\     ] Quotes special characters.  The special characters
               are [$^\.*+?[]].
 
-   Note: the argument to [regexp] is usually a string literal. In this
-   case, any backslash character in the regular expression must be
-   doubled to make it past the OCaml string parser. For example, the
-   following expression:
-   {[ let r = Str.regexp "hello \\([A-Za-z]+\\)" in
-      Str.replace_first r "\\1" "hello world" ]}
+   In regular expressions you will often use backslash characters; it's
+   easier to use a verbatim string literal [{|...|}] to avoid having to
+   escape backslashes. And, if you want a regular expression that
+   matches a literal backslash character, you need to double it, i.e.
+   [Str.regexp {|\\|}].
+
+   For example, the following expression:
+   {[ let r = Str.regexp {|hello \([A-Za-z]+\)|} in
+      Str.replace_first r {|\1|} "hello world" ]}
    returns the string ["world"].
```

Comment:
Could you show the example twice, once with usual string literals and the other with quoted string literals?